### DISCUSSION TITLE
What does the _update_alg( ) method do?

### DISCUSSION THREAD

**Discussion by leo2r** [pyvista/pyvista#3275] [Q&A] [answered]
What exactly does the _update_alg( ) method do? I get a segfault and the fault handler points back to this method.

Error message:

```
Fatal Python error: Segmentation fault

Current thread 0x00000001134af600 (most recent call first):
  File "/Users/leoderiedmatten/miniconda3/envs/t21-pyvista/lib/python3.7/site-packages/pyvista/core/filters/__init__.py", line 36 in _update_alg
  File "/Users/leoderiedmatten/miniconda3/envs/t21-pyvista/lib/python3.7/site-packages/pyvista/core/filters/poly_data.py", line 93 in _boolean
  File "/Users/leoderiedmatten/miniconda3/envs/t21-pyvista/lib/python3.7/site-packages/pyvista/core/filters/poly_data.py", line 253 in boolean_intersection
  File "min_eg.py", line 11 in <module>
Segmentation fault: 11
```

For reference, the code for _update_alg( ) found in /pyvista/core/filters/__init__.py  is:

```python
def _update_alg(alg, progress_bar=False, message=''):
    """Update an algorithm with or without a progress bar."""
    if progress_bar:
        with ProgressMonitor(alg, message=message):
            alg.Update()
    else:
        alg.Update()
```

**Comment by MatthewFlamm** [comment, marked as answer]
I believe this is the method that typically 'executes' the algorithm in use.  Do you know how to use the traceback to find which algorithm was being executed?

Sometimes this indicates a memory overrun, sometimes this indicates that the input is not as expected, e.g. the algorithm expects points on a plane, but you've provided points in 3D space, etc.  It could also be a be a vtk or pyvista bug.

**Reply by leo2r** [reply]
Thanks for the answer, that makes sense! It's happening when I try to calculate the boolean_intersection of two meshes. Unfortunately the fault handler does not give me more information as to where it is actually failing, but it seems to be on the VTK side, probably somewhere in ```vtkBooleanOperationPolyDataFilter```.... it's so frustrating I've tried so many variations to my mesh but this error still happens 😭

**Reply by MatthewFlamm** [reply]
I'm not too familiar with the boolean operations, but I believe that the surface normals are important here.  Maybe others with more direct experience will chime in too.

**Reply by banesullivan** [reply]
Issues with boolean operations come up time and time again... often for this, we point people to external libraries for better support than what VTK has to offer.

@akaszynski has much more experience here than I

**Reply by leo2r** [reply]
Might have found an alternative to using boolean operations thanks to @adeak 's help! So all good for now, thanks :)
